java-checks/src/main/java/org/sonar/java/checks/AbstractPrintfChecker.java

```diff
@@ -222,7 +222,11 @@ protected void reportMissingPrevious(MethodInvocationTree mit) {
   }
 
   protected static Integer getIndex(String param) {
-    return Integer.valueOf(param.substring(0, param.indexOf('$')));
+    try {
+      return Integer.valueOf(param.substring(0, param.indexOf('$')));
+    } catch (Exception e) {
```

Comment:
Any reason not to catch a more precise exception, e.g. `NumberFormatException`?

Reply:
Yes, but not a good one: I just forgot to change it back after I changed it for debugging purposes :man_facepalming: Fixed.